doc/index.md

```diff
@@ -152,6 +152,17 @@ mlpack in C++, but almost all the same algorithms are available.
 | ***Command-line programs*** | -- | [quickstart](quickstart/cli.md) | -- | [reference](user/bindings/cli.md) |
 | ***Go*** | -- | [quickstart](quickstart/go.md) | -- | [reference](user/bindings/go.md) |
 
+## mlpack on embedded systems
```

Comment:
I wonder if this might look more visually organized if it was a subsection of `Examples and further documentation`.  There are a lot of `<h2>`s on the main index page, my eyes get just a little lost looking at them.